### PR TITLE
Add check for necessary requirements

### LINKED ISSUE #60: Dependencies used but not checked for
Namely [dbus-1](https://github.com/FedeDP/Clightd/blob/master/CMakeLists.txt#L116) and [systemd](https://github.com/FedeDP/Clightd/blob/master/CMakeLists.txt#L144). This can lead to cmake errors not easy to trace.

### PR DESCRIPTION
**Pull request by lgbaldoni**
Fixes https://github.com/FedeDP/Clightd/issues/60

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -26,6 +26,10 @@ set_property(TARGET ${PROJECT_NAME} PROPERTY C_STANDARD 99)
 # Required dependencies
 pkg_check_modules(REQ_LIBS REQUIRED libudev libmodule>=5.0.0 libjpeg)
 pkg_check_modules(POLKIT REQUIRED polkit-gobject-1)
+pkg_check_modules(DBUS REQUIRED dbus-1)
+if(WITH_DDC)
```
Comment (FedeDP) [inline CMakeLists.txt:30]: Does ddcutil require systemd? It does not seem so, looking at their configure.ac: https://github.com/rockowitz/ddcutil/blob/master/configure.ac.
DBUS check looks fine, thanks!

Comment (lgbaldoni) [reply CMakeLists.txt:30]: > Does ddcutil require systemd?

See https://github.com/FedeDP/Clightd/blob/master/CMakeLists.txt#L144

Comment (FedeDP) [reply CMakeLists.txt:30]: I see, thanks, didn't notice!

**Comment by FedeDP** [comment]
Indeed i saw that systemd was already required at line124 @lgbaldoni :
> Only install systemd service in systemd environment

I did not add the REQUIRED flag because i don't want to break non-systemd systems.
Is it ok for you?

**Comment by lgbaldoni** [comment]
> I did not add the REQUIRED flag because i don't want to break non-systemd systems.
Is it ok for you?

Right, I was thinking of that.
It's good, at least there's an indication in the log of what could be missing.

**Comment by lgbaldoni** [comment]
> Indeed i saw that systemd was already required at line124

Sorry, were you referring to [libsystemd](https://github.com/FedeDP/Clightd/blob/master/CMakeLists.txt#L121) ?

**Comment by FedeDP** [comment]
> Sorry, were you referring to libsystemd ?

You are right, i didn't even notice!
Anyway, i just pushed some improvements to CMakeLists; hopefully this improves the situation a bit. Let me know please!

**Comment by lgbaldoni** [comment]
> Let me know please!

👍

**Pull request merged**